client/taskenv/services.go

```diff
@@ -36,6 +36,7 @@ func InterpolateServices(taskEnv *TaskEnv, services []*structs.Service) []*struc
 
 		service.Name = taskEnv.ReplaceEnv(service.Name)
 		service.PortLabel = taskEnv.ReplaceEnv(service.PortLabel)
+		service.Address = taskEnv.ReplaceEnv(service.Address)
```

Comment:
Let's add a test to assert this gets interpolated. We rarely touch this, so the risk of it getting accidentally broken is low, but given how critical interpolation is to this feature I think it's worth covering at least once.